lib/client/api.go

```diff
@@ -1663,7 +1663,12 @@ func (tc *TeleportClient) LogoutAll() error {
 // If 'activateKey' is true, saves the received session cert into the local
 // keystore (and into the ssh-agent) for future use.
 //
-func (tc *TeleportClient) Login(ctx context.Context, activateKey bool) (*Key, error) {
+func (tc *TeleportClient) Login(ctx context.Context, activateKey bool) (key *Key, err error) {
+	defer func() {
+		// Wrap error with user friendly message about tsh profile context
+		err = trace.WrapWithMessage(err, "for user %v on %v", tc.Username, tc.WebProxyAddr)
```

Comment:
I'm not sure this is the most readable approach and also not sure this is the responsibility of the API - what if these fields are unset (if that's a possiblity)?
It would be better if the error was augmented at the call site where necessary.

Reply:
Yes, I agree, I need to clear up exactly where we want this error meta data in the issue.